cmd/cli/plugin/package/test/package_plugin_integration_test.go

```diff
@@ -610,12 +628,33 @@ func pauseKappControllerPackage(clusterName, clusterNamespace, mgmtClusterKubeco
 	By("pausing kapp controller app")
 	mgmtClusterKubeCtx := mgmtClusterName + "-admin@" + mgmtClusterName
 
+	var (
+		carvelKind          string
+		packageInstallFound bool
+	)
+
 	command := exec.NewCommand(
 		exec.WithCommand("kubectl"),
-		exec.WithArgs("patch", "app", fmt.Sprintf("%s-kapp-controller", clusterName), "-n", clusterNamespace, "--type", "merge", "-p", fmt.Sprintf("{\"spec\":{\"paused\":%t}}", pause), "--context", mgmtClusterKubeCtx, "--kubeconfig", mgmtClusterKubeconfigPath),
+		exec.WithArgs("get", "packageinstall", fmt.Sprintf("%s-kapp-controller", clusterName), "-n", clusterNamespace, "--context", mgmtClusterKubeCtx, "--kubeconfig", mgmtClusterKubeconfigPath),
 		exec.WithStdout(GinkgoWriter),
 	)
 	err := command.RunAndRedirectOutput(context.Background())
+	if err == nil {
```

Comment:
What if the error be something other than NotFound?

Reply:
Unfortunately the error returned by RunAndRedirectOutput is errors.WithStack and it doesnt return the actual error string. So cannot distinguish between NotFound and other error. The likelihood of other error happening is small as well as this is test code. So I think we can live with this for now and try to enhance it later.

Reply:
Since we have added the hack to create legacy clusters, maybe we can just skip adding this logic to check for PackageInstall or App for now?

Reply:
> Unfortunately the error returned by RunAndRedirectOutput is errors.WithStack and it doesnt return the actual error string. So cannot distinguish between NotFound and other error. The likelihood of other error happening is small as well as this is test code. So I think we can live with this for now and try to enhance it later.

I see, thanks for the clarification

Reply:
> Since we have added the hack to create legacy clusters, maybe we can just skip adding this logic to check for PackageInstall or App for now?

Downstream tests run on classy clusters. So need to add this logic for packageinstall as well as app.